scripts/common/proxy.sh

```diff
@@ -47,6 +47,46 @@ function proxy_bootstrap() {
     fi
 }
 
+# check_proxy_config tries to check if is possible connect with the registry
+# the following code will check if the proxy is invalid by doing a check as following
+# ~$ sudo HTTP_PROXY=http://77.89.35.50:8080 HTTPS_PROXY=http://77.89.35.50:8080 ctr image pull docker.io/library/nginx:latest
+# docker.io/library/nginx:latest: resolving      |--------------------------------------|
+# elapsed: 29.9s                  total:   0.0 B (0.0 B/s)
+# INFO[0030] trying next host                              error="failed to do request: Head \"https://registry-1.docker.io/v2/library/nginx/manifests/latest\": proxyconnect tcp: dial tcp 77.89.35.50:8080: i/o timeout" host=registry-1.docker.io
+# ctr: failed to resolve reference "docker.io/library/nginx:latest": failed to do request: Head "https://registry-1.docker.io/v2/library/nginx/manifests/latest": proxyconnect tcp: dial tcp 77.89.35.50:8080: i/o timeout
+function check_proxy_config() {
+    if [ -n "$CONTAINERD_VERSION" ]; then
+        logStep "Checking proxy configuration with Containerd"
+
+        # Echo containerd Proxy config:
+        local proxy_config_file="/etc/systemd/system/containerd.service.d/http-proxy.conf"
+        if [ -f "$proxy_config_file" ]; then
+            echo ""
+            log "Proxy config:"
+            grep -v -e '^\[Service\]' -e '^# Generated by kURL' "$proxy_config_file"
+            echo ""
+        else
+            log "Skipping test. No HTTP proxy configuration found."
+            return
+        fi
+
+        # Get proxy configuration from http-proxy.conf file
+        local http_proxy=$(grep -E '^\s*Environment\s*=\s*"HTTP_PROXY=' "$proxy_config_file" | sed -E 's/^\s*Environment\s*=\s*"HTTP_PROXY=(.*)"/\1/')
+        local https_proxy=$(grep -E '^\s*Environment\s*=\s*"HTTPS_PROXY=' "$proxy_config_file" | sed -E 's/^\s*Environment\s*=\s*"HTTPS_PROXY=(.*)"/\1/')
+        local no_proxy=$(grep -E '^\s*Environment\s*=\s*"NO_PROXY=' "$proxy_config_file" | sed -E 's/^\s*Environment\s*=\s*"NO_PROXY=(.*)"/\1/')
+
+        # Make the Docker image pull call with proxy configuration
+        if response=$(sudo ctr image pull --http-proxy "$http_proxy" --https-proxy "$https_proxy" --no-proxy "$no_proxy" docker.io/library/nginx:latest 2>&1) && [[ $response == *"proxy"* ]]; then
```

Comment:
is the message from this command helpful in debugging?

Reply:
We are outputting the config and if an issue be identified we are letting the user know that the problem is related with the proxy and we link the docs The docs are not great now. But is were we centralize the info right. so we can improve it in the long run.

Otherwise, what would be your suggestion here?